harvester/records/formats/marc.py

```diff
@@ -20,70 +41,197 @@ class MARC(MarcalyxSourceRecord):
-        raise NotImplementedError
+        return None
 
     def _dct_temporal_sm(self) -> list[str] | None:
-        raise NotImplementedError
+        return None
 
     def _gbl_dateRange_drsim(self) -> list[str]:
-        raise NotImplementedError
+        return []
 
     def _gbl_resourceType_sm(self) -> list[str]:
-        raise NotImplementedError
+        return []
 
     def _gbl_indexYear_im(self) -> list[int]:
-        raise NotImplementedError
+        return []
+
+    ##########################
+    # Helpers
+    ##########################
+
+    def _get_largest_bounding_box(
+        self,
+    ) -> dict[Literal["w", "e", "n", "s"], Decimal] | None:
+        """Method to return largest bounding box from 034 tags.
+
+        Subfield mapping:
+            $d - Coordinates - westernmost longitude (NR)
+            $e - Coordinates - easternmost longitude (NR)
+            $f - Coordinates - northernmost latitude (NR)
+            $g - Coordinates - southernmost latitude (NR)
+        """
+        subfield_to_direction = {"d": "w", "e": "e", "f": "n", "g": "s"}
+
+        # parse and filter 034 tags
+        tags = self.marc.field("034")
+        tags = [
+            tag
+            for tag in tags
+            if all(tag.subfield(subfield) for subfield in subfield_to_direction)
+        ]
+        if not tags:
+            message = "Record does not have valid 034 tag(s), cannot determine bbox."
+            logger.debug(message)
+            return None
+
+        # build dictionary of all corner values
+        bbox_data: dict[str, list] = {
+            direction: [] for direction in subfield_to_direction.values()
+        }
+        for tag in tags:
+            for subfield, direction in subfield_to_direction.items():
+                value = self.convert_coordinate_string_to_decimal(
+                    tag.subfield(subfield)[0].value
+                )
+                if value is not None:
+                    bbox_data[direction].append(value)
+
+        # return None if any four corners do not have values
+        if not all(bbox_data.values()):
+            return None
+
+        # return largest box
+        return {
+            "w": min(bbox_data["w"]),
+            "e": max(bbox_data["e"]),
+            "n": max(bbox_data["n"]),
+            "s": min(bbox_data["s"]),
+        }
+
+    @classmethod
+    def pad_coordinate_string(cls, coordinate_string: COORDINATE_STRING) -> str:
+        """Pad coordinate string with zeros."""
+        hemisphere, coordinate = coordinate_string[0], coordinate_string[1:]
+        if hemisphere in "NSEW":
+            coordinate = f"{coordinate:>07}"
+        return hemisphere + coordinate
+
+    @classmethod
+    def convert_coordinate_string_to_decimal(
+        cls,
+        coordinate_string: COORDINATE_STRING,
+        precision: int = 10,
+    ) -> Decimal | None:
+        """Convert string coordinate to 10 digit precision decimal."""
```

Comment:
Even with the comments, I think this docstring could benefit from a little more detail on why this is done this way